src/robot/libraries/BuiltIn.py

```diff
@@ -2964,6 +2964,8 @@ def log_to_console(self, message, stream='STDOUT', no_newline=False):
         This keyword does not log the message to the normal log file. Use
         `Log` keyword, possibly with argument ``console``, if that is desired.
         """
+        format = "{:" + format + "}"
+        message = format.format(message)
```

Comment:
1. It would be a bit safer to add `if format:` before these two lines. It ought to be always safe to call `{:}.format(message)`, but because `message` can be anything, not only a string, there's a small risk it could blow up.
2. This new functionality needs documentation. The docs should contain a link to https://docs.python.org/3/library/string.html#formatspec and a new example should be added.
3. Docs also need a note that `format` is new in Robot Framework 5.0.